### PR TITLE
6.5.0: TCP stream handler and Sumo Logic metrics handler

### LINKED ISSUE #3357: TCPStreamHandler resource
### Description
Sensu Go 6.5.0 will include a new handler resource, the TCPStreamHandler (enterprise/v1.TCPStreamHandler).

### Category

This is an issue with:

- [ ] errata (error in doc content)
- [x] new doc (requires a completely new doc)
- [ ] update (add missing or refresh existing content)
- [ ] improvement (improve docs content or format)

### Corrections needed
- Add API doc and reference page for TCPStreamHandler
- Update pipelines and handlers reference pages to explain TCPStreamHandler

### Additional context
Issue: https://github.com/sensu/sensu-enterprise-go/issues/1695
PRs: https://github.com/sensu/sensu-enterprise-go/pull/1770 and https://github.com/sensu/sensu-enterprise-go/pull/1780
also https://github.com/sensu/sensu-staging/pull/286/files

### LINKED ISSUE #3358: SumoLogicMetricsHandler resource
### Description
Sensu Go 6.5.0 will include a new handler resource, the SumoLogicMetricsHandler (enterprise/v1.SumoLogicMetricsHandler).

### Category

This is an issue with:

- [ ] errata (error in doc content)
- [x] new doc (requires a completely new doc)
- [ ] update (add missing or refresh existing content)
- [ ] improvement (improve docs content or format)

### Corrections needed
- Add API doc and reference page for SumoLogicMetricsHandler
- Update pipelines and handlers reference pages to explain SumoLogicMetricsHandler

### Additional context
Issue: https://github.com/sensu/sensu-enterprise-go/issues/1696

No PR yet

### PR DESCRIPTION
**Pull request by hillaryfraley**
## Description
- Adds API and reference docs for TCP stream handlers and Sumo Logic metrics handler.
- Adds usage notes to explain the difference between core/.../handlers and enterprise/handler/v1 resources

## Motivation and Context
Closes https://github.com/sensu/sensu-docs/issues/3357
Closes https://github.com/sensu/sensu-docs/issues/3358

### CODE REVIEW

**Comment by hillaryfraley** [comment]
**NEEDED**
- Are SumoLogicMetricsHandler and TCPStreamHandler supported for sensuctl edit or sensuctl prune?

File: content/sensu-go/6.5/sensuctl/back-up-recover.md
Comment (amdprophet) [inline content/sensu-go/6.5/sensuctl/back-up-recover.md:353]: Missing `core/v2.Pipeline`.

File: content/sensu-go/6.5/sensuctl/create-manage-resources.md
Comment (amdprophet) [inline content/sensu-go/6.5/sensuctl/create-manage-resources.md]: Should be `core/v2.Pipeline` for the fully qualified name and `Pipeline` for type.

**Review by amdprophet — CHANGES_REQUESTED** [review]

**Review by amdprophet — APPROVED** [review]

**Comment by portertech** [comment]
@hillaryfraley We need to be more specific in regards to the Sumo HTTP Source configuration to be able to know how many Sensu Go users are sending data to Sumo (we can't just link to their docs). We need users to add a custom field to their source, I'm working on a name/value for this :sweat_smile:

**Comment by portertech** [comment]
@hillaryfraley we've merged https://github.com/sensu/sensu-enterprise-go/pull/1817 which gets us our most reliable means of generating a Sensu+ adoption report. There's ongoing discussion in regards to documenting a user/customer defined custom field in their HTTP source, it's another step that some want to avoid. Don't let this block this pull-request, we can add/adjust accordingly at a later time.

File: content/sensu-go/6.5/observability-pipeline/observe-process/sumo-logic-metrics-handlers.md
Comment (fguimond) [inline content/sensu-go/6.5/observability-pipeline/observe-process/sumo-logic-metrics-handlers.md:6]: ~Maybe it should be `executes on event metrics` instead of just events.~

File: content/sensu-go/6.5/observability-pipeline/observe-process/sumo-logic-metrics-handlers.md
Comment (fguimond) [inline content/sensu-go/6.5/observability-pipeline/observe-process/sumo-logic-metrics-handlers.md]: This was actually not possible with the HTTP connection pool. The only parameters are

- maximum number of connections
- timeout - the maximum time allowed for a request to complete

File: content/sensu-go/6.5/observability-pipeline/observe-process/sumo-logic-metrics-handlers.md
Comment (fguimond) [inline content/sensu-go/6.5/observability-pipeline/observe-process/sumo-logic-metrics-handlers.md]: There is no minimum, the connections will terminate automatically when idle for too long (more than 2 minutes). This should probably be something like `with a pool of up to 10 connections`.

File: content/sensu-go/6.5/observability-pipeline/observe-process/sumo-logic-metrics-handlers.md
Comment (fguimond) [reply content/sensu-go/6.5/observability-pipeline/observe-process/sumo-logic-metrics-handlers.md:6]: Just saw the note below about metrics, this change is not necessary.

File: content/sensu-go/6.5/observability-pipeline/observe-process/sumo-logic-metrics-handlers.md
Comment (fguimond) [inline content/sensu-go/6.5/observability-pipeline/observe-process/sumo-logic-metrics-handlers.md:46]: Here's an example with the connection pool parameters. It also has secrets which are mentioned further down. 
```
type: SumoLogicMetricsHandler
api_version: pipeline/v1
metadata:
  name: sumologic_http_log_metrics
  namespace: default
spec:
  url: $SUMO_LOGIC_SOURCE_ADDRESS
  secrets:
  - name: SUMO_LOGIC_SOURCE_ADDRESS
    secret: sumo-logic-source-address
  max_connections: 10
  timeout: 30s
```

File: content/sensu-go/6.5/observability-pipeline/observe-process/sumo-logic-metrics-handlers.md
Comment (fguimond) [inline content/sensu-go/6.5/observability-pipeline/observe-process/sumo-logic-metrics-handlers.md]: See comments above re. connection pool attributes.

File: content/sensu-go/6.5/observability-pipeline/observe-process/sumo-logic-metrics-handlers.md
Comment (fguimond) [inline content/sensu-go/6.5/observability-pipeline/observe-process/sumo-logic-metrics-handlers.md:276]: It seems like the `secrets` attribute is a little bit misleading since the actual $SUMOLOGIC_METRICS_URL is not used.

File: content/sensu-go/6.5/observability-pipeline/observe-process/sumo-logic-metrics-handlers.md
Comment (fguimond) [inline content/sensu-go/6.5/observability-pipeline/observe-process/sumo-logic-metrics-handlers.md]: Would it make sense to show the url attribute using the secrets? Something like: 
```
url: $SUMOLOGIC_METRICS_URL
```
I'm not sure how it is documented elsewhere so I'll leave it up to you :)

File: content/sensu-go/6.5/observability-pipeline/observe-process/sumo-logic-metrics-handlers.md
Comment (fguimond) [inline content/sensu-go/6.5/observability-pipeline/observe-process/sumo-logic-metrics-handlers.md]: not supported.

File: content/sensu-go/6.5/observability-pipeline/observe-process/sumo-logic-metrics-handlers.md
Comment (fguimond) [inline content/sensu-go/6.5/observability-pipeline/observe-process/sumo-logic-metrics-handlers.md]: not supported

File: content/sensu-go/6.5/observability-pipeline/observe-process/sumo-logic-metrics-handlers.md
Comment (fguimond) [inline content/sensu-go/6.5/observability-pipeline/observe-process/sumo-logic-metrics-handlers.md]: not supported

File: content/sensu-go/6.5/observability-pipeline/observe-process/sumo-logic-metrics-handlers.md
Comment (fguimond) [inline content/sensu-go/6.5/observability-pipeline/observe-process/sumo-logic-metrics-handlers.md]: also need to have `timeout`

File: content/sensu-go/6.5/observability-pipeline/observe-process/sumo-logic-metrics-handlers.md
Comment (fguimond) [inline content/sensu-go/6.5/observability-pipeline/observe-process/sumo-logic-metrics-handlers.md]: This is required.

File: content/sensu-go/6.5/observability-pipeline/observe-process/sumo-logic-metrics-handlers.md
Comment (fguimond) [inline content/sensu-go/6.5/observability-pipeline/observe-process/sumo-logic-metrics-handlers.md]: Not required. But if set to 0 there is no limit to the pool.

File: content/sensu-go/6.5/observability-pipeline/observe-process/sumo-logic-metrics-handlers.md
Comment (fguimond) [inline content/sensu-go/6.5/observability-pipeline/observe-process/sumo-logic-metrics-handlers.md]: This is a duration expression. Complies with https://pkg.go.dev/time#ParseDuration

**Review by echlebek — COMMENTED** [review]
So, I think something needs to be clarified in the docs, just from reading the highlighted note about the pipeline API.

What you're referring to as the pipeline API is actually an _API group_, a container for APIs. The pipeline/v1 API group contains the TCPStreamHandler and SumoLogicHandler APIs. I think if we adjust the wording accordingly, the concepts should fall into place a bit more easily.

**Comment by hillaryfraley** [comment]
Thanks Eric. We will do a bit more to organize the API docs for the 6.6 release, I think.

File: content/sensu-go/6.5/observability-pipeline/observe-process/tcp-stream-handlers.md
Comment (echlebek) [inline content/sensu-go/6.5/observability-pipeline/observe-process/tcp-stream-handlers.md]: This field does not exist in the implementation. It did exist in the initial spec, but was removed after some consideration.

File: content/sensu-go/6.5/observability-pipeline/observe-process/tcp-stream-handlers.md
Comment (echlebek) [inline content/sensu-go/6.5/observability-pipeline/observe-process/tcp-stream-handlers.md]: Should be edited to reflect that there is no configurable minimum.

**Review by echlebek — APPROVED** [review]
Looks good modulo my comments! :shipit:

**Pull request merged**
